### PR TITLE
#159361806 effectively onboard vendor

### PR DESCRIPTION
**Pull request by ozimos**
#### What does this PR do?
Effectively onboard vendor on to the marketplace application
#### Description of Task to be completed?
- Provide vendor with an overview of the dashboard feature and other homepage admin features
- Provide vendor with a tour of how to add a product to the store
- Enable autostart of the tour on the first visit to landing page or product details page
#### How should this be manually tested?
Visit the review app below, enable the marketplace for the application and then sign up as a vendor. The tour can be started from the navbar button
#### Any background context you want to provide?
NA
#### What are the relevant pivotal tracker stories?
#159361806
#### Screenshots or gifs (if appropriate):
NA
#### Questions:
NA

### CODE REVIEW

File: imports/plugins/custom/completed-orders/client/containers/customMainDropdown.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ import { getRawComponent,
 import { Reaction, Logger }  from "/client/api";
 const MainDropdownComponent = getRawComponent("MainDropdown");
 
+import onboard from "/imports/plugins/custom/vendor-onboarding/client/onboard";
+import { dashboardSteps } from "/imports/plugins/custom/vendor-onboarding/client/tourSteps";
```
Comment (otseobande) [inline imports/plugins/custom/completed-orders/client/containers/customMainDropdown.js]: Shouldn't these imports come before the `MainDropdownComponent` variable assignment?

File: imports/plugins/custom/gdr-landing-page/client/components/CustomLandingPage.js [hunk 2]
```diff
@@ -9,6 +9,9 @@ import Slider  from "./Slider";
 import TrendingProducts from "./TrendingProducts";
 
 const Products = getRawComponent("Products");
+import onboard from "/imports/plugins/custom/vendor-onboarding/client/onboard";
```
Comment (otseobande) [inline imports/plugins/custom/gdr-landing-page/client/components/CustomLandingPage.js]: Like stated above, I think imports should come before other parts of the module

File: imports/plugins/custom/gdr-landing-page/client/components/CustomLandingPage.js [hunk 3]
```diff
@@ -18,16 +21,23 @@ class CustomLandingPage extends Products {
...
+  initializeTour() {
+    setTimeout(() => {
+      onboard.autoDashboardTour(dashboardSteps);
+    }, 2000);
+  }
   getTrendingProducts(limit) {
     Meteor.call("product/getTrendingProducts", limit, (err, res) => {
       if (err) {
-        console.log(err);
+        return err;
```
Comment (otseobande) [inline imports/plugins/custom/gdr-landing-page/client/components/CustomLandingPage.js]: Returning an error here doesn't really do justice to it. Use Meteor's `Logger` with `import { Logger } from "/client/api";` here.

**Review by iverenshaguy — APPROVED** [review]
Nice work! Well done.

**Comment by ozimos** [comment]
@otseobande I have fixed the errors you pointed out. Back to you

File: imports/plugins/custom/vendor-onboarding/client/tourSteps.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    intro: `<h4>Vendor Dashboard</h4>
+        <hr>
+        <div>The dashboard allows you to manage your store, keep up with orders, choose your preferred payment methods and lots more</div>`,
+    position: "left"
+  },
+  {
+    element: ".fa-sun-o",
+    intro: `<h4> Orders </h4>
+        <hr>
+        <div> The orders section shows all the products ordered from yor store.</div>`,
```
Comment (Stylll) [inline imports/plugins/custom/vendor-onboarding/client/tourSteps.js]: Please change "yor" to "your" here.

**Review by Stylll — COMMENTED** [review]
Hi Moses. Great job implementing the tour and feedback received from the team. I noticed when I start the tour as a new vendor (where the tour pops up itself), once I'm done with the first tour, I don't see the button to go to the "How to add product" tour. Please try it out at your end to confirm this. Also, I left a comment on the tourSteps.js file, line 18. Please check it out.

File: imports/plugins/custom/vendor-onboarding/client/tourSteps.js (hunk 4 above)
Comment (ozimos) [reply imports/plugins/custom/vendor-onboarding/client/tourSteps.js]: Thank you for your review. I have fixed both issues now

**Review by Stylll — APPROVED** [review]
Good Job. I see you've made the changes I requested for. Keep it up.

**Comment by rovilay** [comment]
Hi Moses,
You have done a very good job. Nice work on the feature. But I noticed that some information is missing from the product cards, like the `like icon`, the `product owner's name` and for some the `product name` itself. Please check it out. Thanks.

**Review by rovilay — APPROVED** [review]
Very Good work. Keep up the good work.

**Pull request merged**